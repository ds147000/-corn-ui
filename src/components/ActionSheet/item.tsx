import React from 'react'
import { View } from '@tarojs/components'
import { ActionSheetItemProps } from './interface'

export const ActionSheetItem: React.FC<ActionSheetItemProps> = ({ text, align, onClick, suffix }) => {
  return (
    <View className="xrk-actionsheet-item" onClick={onClick}>
      <View className={`xrk-as-text xrk-actionsheet-${align}`}>
        {text}
      </View>
      {suffix}
    </View>
  )
}
