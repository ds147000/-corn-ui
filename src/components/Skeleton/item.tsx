/* eslint-disable no-magic-numbers */
import React from 'react'
import { View } from '@tarojs/components'
import { transformRem } from '../../utils'

export interface SkeletonItemProps {
  /** 宽度，自动转换合适750单位。 默认 690 */
  width?: number
  /** 高度，自动转换合适750单位。 默认 46 */
  height?: number
  /** 顶部外边度，自动转换合适750单位。 默认 15 */
  top?: number
  /** 底部外边度，自动转换合适750单位。 默认 15 */
  bottom?: number
  /** 左部外边度，自动转换合适750单位。 默认 30 */
  left?: number
  /** 右部外边度，自动转换合适750单位。 默认 30 */
  right?: number
}

export const SkeletonItem: React.FC<SkeletonItemProps> = ({
  width = 690,
  height = 46,
  top = 15,
  bottom = 15,
  left = 30,
  right = 30
}) => {
  const _width = transformRem(width)
  const _height = transformRem(height)
  const _top = transformRem(top)
  const _left = transformRem(left)
  const _bottom = transformRem(bottom)
  const _right = transformRem(right)
  const _borderRadius = height > 46 ? transformRem(15) : transformRem(8)

  return (
    <View style={{ paddingTop: _top, paddingBottom: _bottom, paddingLeft: _left, paddingRight: _right }} >
      <View className="xrk-skeleton-item" style={{ width: _width, height: _height, borderRadius: _borderRadius }} />
    </View>
  )
}
