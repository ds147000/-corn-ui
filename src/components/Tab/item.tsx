import React, { useMemo } from 'react'
import ClassNames from 'classnames'
import { View } from '@tarojs/components'
import Image from '../Image'
import Link from '../Link'
import type { TabItemPorps } from './index'

export const Item: React.FC<TabItemPorps> = ({ icon, active, title, onClick, message, url }) => {

  const _class = useMemo(() => ClassNames(
    'xrk-tab-item __x-st-exp __x-st-click',
    active && 'xrk-tab-item-active'
  ), [ active ])

  const content = (
    <View className={message ? 'xrk-tab-item-message' : ''}>
      {icon ? <Image src={icon} className="xrk-tab-item-icon" mode="scaleToFill" imgProps={{ alt: title }} /> : title}
    </View>
  )

  if (url) {
    return (
      <Link to={url} className={_class} onClick={onClick} data-testid="tab-item" target="View" >{content}</Link>
    )
  }

  return (
    <View className={_class} onClick={onClick} data-testid="tab-item" >{content}</View>
  )
}
