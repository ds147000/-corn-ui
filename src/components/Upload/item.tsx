// #if _APP === 'weapp'
import Taro from '@tarojs/taro'
// #endif
import classNames from 'classnames'
import { View } from '@tarojs/components'
import React, { useMemo } from 'react'
import { Upload } from './base'
import Image from '../Image'
import Icon from '../Icon'


interface UploadItemProps {
  onPreview(item: Upload.Media): void
  onRemove?: Upload.onRemove
  /** 图片项类名 */
  className?: string
  item: Upload.Media
}

export const UploadItem: React.FC<UploadItemProps> = ({
  onPreview,
  onRemove,
  className,
  item
}) => {
  const _itemClass = useMemo(() => classNames(
    'corn-upload-item',
    className
  ), [ className ])

  const status = useMemo(() => {
    switch(item.status) {
      case 'loading':
        return <View className="corn-upload-loading" />

      case 'error':
        return <View className="corn-upload-error corn-f corn-ac corn-jc" >上传失败</View>

      default:
        return null
    }
  }, [ item.status ])

  const _onRemove = (): void => {
    onRemove?.(item)
  }

  const _onPreview = (): void => {
    onPreview(item)
  }

  return (
    <View className={_itemClass} data-testid="upload-item" >
      <View
        className="corn-upload-remove corn-f corn-ac corn-jc"
        onClick={_onRemove}
        data-testid="upload-remove"
      >
        <Icon name="delete"  />
      </View>
      <Image
        className="corn-upload-img"
        src={item.content}
        onClick={_onPreview}
        lazyLoad
        mode="aspectFill"
        // #if _APP === 'weapp'
        // eslint-disable-next-line no-magic-numbers
        style={{ width: Taro.pxTransform(160), height: Taro.pxTransform(160) }}
        // #endif
      />
      {status}
    </View>
  )
}
