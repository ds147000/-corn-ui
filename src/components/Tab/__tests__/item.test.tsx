import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import { Item } from '../item'

describe('comps/Tab/item', () => {

  test('title', () => {
    const screen = render(<Item title="item" />)
    screen.getByText('item')
    expect(screen.container).toMatchSnapshot()
  })

  test('icon', () => {
    const screen = render(<Item icon="https://t7.baidu.com/it/u=1951548898,3927145&fm=193&f=GIF" />)
    expect(screen.container.querySelector('img')?.src)
      .toBe('https://t7.baidu.com/it/u=1951548898,3927145&fm=193&f=GIF')

    expect(screen.container).toMatchSnapshot()
  })

  test('active', () => {
    const screen = render(<Item title="item" active />)
    expect(screen.getByText('item').className.indexOf('active'))
    expect(screen.container).toMatchSnapshot()
  })

  test('message', () => {
    const screen = render(<Item title="item" message />)
    // eslint-disable-next-line no-magic-numbers
    expect(screen.getByText('item').className.indexOf('xrk-tab-item-message')).not.toBe(-1)
    expect(screen.container).toMatchSnapshot()
  })

  test('click', () => {
    const click = jest.fn()
    const screen = render(<Item title="item" onClick={click} />)
    fireEvent.click(screen.getByText('item'))

    // eslint-disable-next-line no-magic-numbers
    expect(click).toHaveBeenCalledTimes(1)
  })
})
