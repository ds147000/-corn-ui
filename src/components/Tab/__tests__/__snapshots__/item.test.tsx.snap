// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comps/Tab/item active 1`] = `
<div>
  <div
    class="xrk-tab-item __x-st-exp __x-st-click xrk-tab-item-active"
    data-testid="tab-item"
  >
    <div
      class=""
    >
      item
    </div>
  </div>
</div>
`;

exports[`comps/Tab/item icon 1`] = `
<div>
  <div
    class="xrk-tab-item __x-st-exp __x-st-click"
    data-testid="tab-item"
  >
    <div
      class=""
    >
      <div
        class="taro-img xrk-img xrk-tab-item-icon"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://t7.baidu.com/it/u=1951548898,3927145&fm=193&f=GIF"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`comps/Tab/item message 1`] = `
<div>
  <div
    class="xrk-tab-item __x-st-exp __x-st-click"
    data-testid="tab-item"
  >
    <div
      class="xrk-tab-item-message"
    >
      item
    </div>
  </div>
</div>
`;

exports[`comps/Tab/item title 1`] = `
<div>
  <div
    class="xrk-tab-item __x-st-exp __x-st-click"
    data-testid="tab-item"
  >
    <div
      class=""
    >
      item
    </div>
  </div>
</div>
`;
